src/core/internal/dassert.d

```diff
@@ -88,7 +88,7 @@ private string miniFormat(V)(const ref V v)
     else static if (__traits(isFloating, V))
     {
         char[60] val;
-        const len = sprintf(&val[0], "%g", v);
+        const len = sprintf(&val[0], "%Lg", cast(real) v);
```

Comment:
```suggestion
        const len = sprintf(&val[0], "%Lg", real(v));
```

Also not sure about how I feel about everything being upcast to `real`.

Reply:
Fixed.

Reply:
With your suggestion :

> Error: cannot implicitly convert expression v of type const(ifloat) to real

Will this cast change the result of `len`, depending of the base type ?

Reply:
you could just enumerate the different floating point types.

Reply:
@Luhrel : Can you add a real for `real` and the various imaginary types [here](https://github.com/dlang/druntime/blob/b57ca7c86005e1e33cac1623c10b1a2fb6e2adc7/test/exceptions/src/assert_fail.d#L62-L67) ?

Reply:
> you could just enumerate the different floating point types.

It won't work with `float`, because `printf` doesn't accept it.
I simply escape the check.



> @Luhrel : Can you add a real for `real` and the various imaginary types [here](https://github.com/dlang/druntime/blob/b57ca7c86005e1e33cac1623c10b1a2fb6e2adc7/test/exceptions/src/assert_fail.d#L62-L67) ?

Done.